ts/components/core/selection/CheckBox.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from "react";
+import { useState } from "react";
+import { StyleSheet } from "react-native";
+import TouchableDefaultOpacity from "../../TouchableDefaultOpacity";
+import IconFont from "../../ui/IconFont";
+import { calculateSlop } from "../accessibility";
+import { IOColors } from "../variables/IOColors";
+
+type Props = {
+  // the value of the checkbox
+  checked?: boolean;
+  // dispatch the new value after the checkbox changes state
+  onValueChange?: (newValue: boolean) => void;
+};
+
+const size = 24;
+
+const offColor = IOColors.bluegrey;
+const onColor = IOColors.blue;
+const slop = calculateSlop(size);
+const tickSize = size * 0.85;
+
+const styles = StyleSheet.create({
+  checkBox: {
+    width: size,
+    height: size,
+    backgroundColor: IOColors.white,
+    borderColor: IOColors.blue,
+    borderWidth: 2,
+    borderRadius: 4
+  }
+});
+
+const getBorderColor = (value: boolean) => (value ? onColor : offColor);
+
+/**
+ * A checkbox that follows the style guidelines
+ * The graphical size of the element is 24x24 but the touchable area is 48x48
+ *
+ * https://pagopa.invisionapp.com/console/IO-app---Library-ckcdil0710mt1014buxuo4u34/ckcdilqwl032g01xbh77f4i2m/play
+ *
+ * @param props
+ * @constructor
+ */
+export const CheckBox: React.FunctionComponent<Props> = props => {
+  const [toggleValue, setToggleValue] = useState(props.checked ?? false);
+
+  return (
+    <TouchableDefaultOpacity
+      hitSlop={{ bottom: slop, left: slop, right: slop, top: slop }}
+      style={[styles.checkBox, { borderColor: getBorderColor(toggleValue) }]}
+      onPress={() => {
+        setToggleValue(!toggleValue);
```

Comment:
Could be an improvement if this component doesn't set the state by it self?
There are some cases where who uses this component wants to know if the component is pressed (giving the negate of the current state). Then some other logic can apply that change or not.

ex: use the checkbox to make a remote operation. When the remote update starts I would disable the component and set the new state when that operation completes.

Reply:
a20f46d 
Added `RawCheckBox`, a stateless dummy checkbox that accepts only the `checked` value and dispatch the `onPress`.
Changed `CheckBox` in order to use the `RawCheckBox` with a state management.
In this way we can have a "standard" checkbox with the classic state management and a raw checkbox that can be used to implements any particular case like the remote operation.